ultralytics/data/annotator.py

```diff
@@ -64,3 +71,337 @@ def auto_annotate(
+        process: Performs inference and returns bounding boxes and labels in YOLO-compatible format.
+    """
+
+    def __init__(self, device="", variant="base"):
+        """
+        Initializes the Florence2 model with the specified variant and device.
+
+        Args:
+            device (str): Target device for inference, e.g., 'cuda', 'cpu'.
+            variant (str): Model variant to use: "base", "large", "base-ft", "large-ft".
+        """
+        self.device = select_device(device)
+
+        supported_variants = {"base", "large", "large-ft", "base-ft"}
+        self.variant = variant if variant in supported_variants else "base"
+
+        if variant not in supported_variants:
+            LOGGER.warning(
+                f"⚠️ Invalid variant '{variant}' provided. Falling back to 'base'. "
+                f"Supported variants: {supported_variants}"
+            )
+
+        self.mid = f"microsoft/florence-2-{self.variant}"
+        self.model = None
+        self.processor = None
+        self._load_model()
+
+    def _load_model(self):
+        """
+        Loads the Florence2 model and its corresponding processor from Hugging Face Hub.
+
+        This method requires `transformers` and `einops`. It sets up the model and processor
+        using `trust_remote_code=True`, which allows Florence2 custom architecture loading.
+
+        Returns:
+            None
+        """
+        from ultralytics.utils.checks import check_requirements
+
+        check_requirements(["transformers==4.49.0", "einops"])
+
+        from transformers import AutoModelForCausalLM, AutoProcessor, logging
+
+        logging.set_verbosity_error()  # Minimize Hugging Face logs
+
+        LOGGER.info(f"💡 Initializing Florence2-{self.variant} on {str(self.device).upper()}")
+
+        self.model = AutoModelForCausalLM.from_pretrained(
+            self.mid,
+            trust_remote_code=True,
+        ).to(self.device)
+
+        self.processor = AutoProcessor.from_pretrained(
+            self.mid,
+            trust_remote_code=True,
+        )
+
+        LOGGER.info(f"🚀 Loaded Florence2-{self.variant} model successfully.")
+
+    def process(self, im0, w, h):
+        """
+        Performs object detection using Florence2 on an input image.
+
+        This method sends a grounding-style prompt "<OD>" to the model to detect objects
+        and returns predictions formatted for YOLO-compatible annotation.
+
+        Args:
+            im0 (np.ndarray): Input image (HWC format).
+            w (int): Image width.
+            h (int): Image height.
+
+        Returns:
+            result (dict): Dictionary with keys "bboxes" and "labels", formatted for YOLO annotation.
+                  Structure:
+                  {
+                      "bboxes": List of [x1, y1, x2, y2] floats,
+                      "labels": List of str or int label names
+                  }
+
+        Example:
```

Comment:
@RizwanMunawar "Example:" this is incorrect for Google-style docstrings. Please review https://docs.ultralytics.com/help/contributing/#google-style-docstrings